enterprise/com/src/main/java/org/neo4j/com/storecopy/StoreCopyClient.java

```diff
@@ -281,6 +283,20 @@ public boolean visit( CommittedTransactionRepresentation transaction ) throws IO
             // header of the log that we just wrote.
             File currentLogFile = logFiles.getLogFileForVersion( currentLogVersion );
             writeLogHeader( fs, currentLogFile, currentLogVersion, max( BASE_TX_ID, endTxId - 1 ) );
+
+            if ( !forensics )
+            {
+                // since we just create new log and put checkpoint into it with offset equals to
+                // LOG_HEADER_SIZE we need to update last transaction offset to be equal to this newly defined max
+                // offset otherwise next checkpoint that use last transaction offset will be created for non
+                // existing offset that is in most of the cases bigger than new log size.
+                // Recovery will treat that as last checkpoint and will not try to recover store till new
+                // last closed transaction offset will not overcome old one. Till that happens it will be
+                // impossible for recovery process to restore the store
+                File neoStore = new File( tempStoreDir, NeoStore.DEFAULT_NAME );
+                NeoStore.setRecord( fs, neoStore, NeoStore.Position.LAST_CLOSED_TRANSACTION_LOG_BYTE_OFFSET,
```

Comment:
@tinwelint  answering your question:
currently new copied store will reuse same version as original store.
* in case of `forensics == false` since we do not copy logs we will create new file for the same log version which i suppose is fine
* for `forensics == true` indeed since log files will be copied we will use corresponding log file with new store as well, and checkpoint that we just create at 0 does not make a lot of sense.

We can increment neostore log version here and reset transaction offset to the the first checkpoint. That will change current `forensics` way of working which probably should make identical copy of initial store. But if we can change that, will agree that also incrementing store version should make more sense.